src/wallet3/wallet.cpp

```diff
@@ -185,11 +182,24 @@ namespace wallet
       return;
     }
 
+    std::vector<std::thread> threads;
+
+    auto db_tx = db->db_transaction();
     for (const auto& block : blocks)
```

Comment:
if we could `std::move` the block in this foreach into that lambda, we can capture `[this, block=std::move(block)]` we can avoid a capture all by reference (which is something i always try to avoid using). that or capture with `[this, &block]`

Reply:
also the insides of `add_block` sound possibly thread unsafe. 

do we mark all functions that are pure with `const` or no?